File: golang/3014_minimum_pushes/minimum_pushes.go
// Package main ...
package main

import "fmt"

func minimumPushes(word string) int {
	n := len(word)
	// 向上取整
	m := (n-1)/8 + 1
	return (m-1+1)*(m-1)/2*8 + (n-(m-1)*8)*m
}

func main() {
	tests := []struct{
		word string
		ans int
	}{
      {"abcde", 5},
      {"xycdefghij", 12},
	}

	for index, test := range tests {
		if minimumPushes(test.word) != test.ans {
			fmt.Println(index)
		}
	}
}

File: rust/3016_minimum_pushes/minimum_pushes.rs
struct Solution;

impl Solution {
    pub fn minimum_pushes(word: String) -> i32 {
        let mut freq = vec![0; 26];
        for c in word.chars() {
            freq[(c as u8 - b'a') as usize] += 1;
        }

        freq.sort_by(|a, b| b.cmp(a));

        let mut ans = 0;
        for i in 0..26 {
            if freq[i] == 0 {
                break;
            }
            ans += (freq[i] * (i / 8 + 1)) as i32;
        }
        ans
    }
}

fn main() {
    let tests = vec![
        ("abcde", 5),
        ("xyzxyzxyzxyz", 12),
        ("aabbccddeeffgghhiiiiii", 24),
    ];

    for (word, expected) in tests {
        assert_eq!(Solution::minimum_pushes(word.to_string()), expected);
    }
}
